provider/aws/formation/rack.json

```diff
@@ -2713,47 +2676,102 @@
       "Properties": {
-        "ConnectionDrainingPolicy": { "Enabled": true, "Timeout": 60 },
-        "ConnectionSettings": { "IdleTimeout": 3600 },
-        "CrossZone": true,
-        "HealthCheck": {
-          "HealthyThreshold": "2",
-          "Interval": 5,
-          "Target": { "Fn::If": [ "PrivateApi", "HTTPS:3100/check", "HTTPS:3000/check" ] },
-          "Timeout": 3,
-          "UnhealthyThreshold": "2"
-        },
-        "LBCookieStickinessPolicy": [ { "PolicyName": "affinity" } ],
-        "Listeners": { "Fn::If": [ "PrivateApi",
-          [
-            { "Protocol": "TCP", "LoadBalancerPort": "443", "InstanceProtocol": "TCP", "InstancePort": "3100" }
-          ],
-          [
-            { "Protocol": "TCP", "LoadBalancerPort": "443", "InstanceProtocol": "TCP", "InstancePort": "3000" }
-          ]
-        ] },
-        "LoadBalancerName": { "Fn::If": [ "PrivateApi", { "Fn::Join": [ "-", [ { "Ref": "AWS::StackName" }, "i" ] ] }, { "Ref": "AWS::StackName" } ] },
-        "Scheme": { "Fn::If": [ "PrivateApi", "internal", { "Ref": "AWS::NoValue" } ] },
-        "SecurityGroups": [ { "Ref": "BalancerSecurity" } ],
+        "Name": { "Fn::If": [ "PrivateApi", { "Fn::Join": [ "-", [ { "Ref": "AWS::StackName" }, "i" ] ] }, { "Ref": "AWS::StackName" } ] },
+        "LoadBalancerAttributes": [
+          { "Key": "access_logs.s3.enabled", "Value": "true" },
+          { "Key": "access_logs.s3.bucket", "Value": { "Fn::If": [ "BlankLogBucket", { "Ref": "Logs" }, { "Ref": "LogBucket" } ] } },
+          { "Key": "access_logs.s3.prefix", "Value": { "Fn::Sub": "convox/logs/${AWS::StackName}/alb" } },
+          { "Key": "idle_timeout.timeout_seconds", "Value": { "Ref": "LoadBalancerIdleTimeout" } },
+          { "Key": "routing.http.desync_mitigation_mode", "Value": { "Ref": "RouterMitigationMode" } },
+          { "Key": "routing.http.drop_invalid_header_fields.enabled", "Value": "true" }
+        ],
+        "Scheme": { "Fn::If": [ "PrivateApi", "internal", "internet-facing" ] },
         "Subnets": {
           "Fn::If": [ "PrivateApi",
-            [ { "Ref": "SubnetPrivate0" }, { "Ref": "SubnetPrivate1" }, { "Fn::If": [ "ThirdAvailabilityZoneAndHighAvailability", { "Ref": "SubnetPrivate2" }, { "Ref": "AWS::NoValue" } ] } ],
+            [ { "Ref": "SubnetPrivate0" }, { "Ref": "SubnetPrivate1" }, { "Fn::If": [ "CreateThirdPrivateSubnet", { "Ref": "SubnetPrivate2" }, { "Ref": "AWS::NoValue" } ] } ],
             [ { "Ref": "Subnet0" }, { "Ref": "Subnet1" }, { "Fn::If": [ "ThirdAvailabilityZoneAndHighAvailability", { "Ref": "Subnet2" }, { "Ref": "AWS::NoValue" } ] } ]
           ]
         },
+        "SecurityGroups": [ { "Ref": "ApiBalancerSecurity" } ],
         "Tags": [
-          { "Key": "GatewayAttachment", "Value": { "Fn::If": [ "ExistingVpc", "existing", { "Ref": "GatewayAttachment" } ] } },
-          { "Key": "Name", "Value": { "Ref": "AWS::StackName" } }
+          { "Key": "Rack", "Value": { "Ref": "AWS::StackName" } }
```

Comment:
Should we leave the current tags?

Reply:
I think we can leave as it just to keep consistency with the other ALB https://github.com/convox/rack/blob/switch-elb/provider/aws/formation/rack.json#L2616